store/storetest/shared_channel_store.go

```diff
@@ -783,3 +802,152 @@ func testUpdateSharedChannelUserLastSyncAt(t *testing.T, ss store.Store) {
 		require.Error(t, err, "update non-existent user should error", err)
 	})
 }
+
+func testSaveSharedChannelAttachment(t *testing.T, ss store.Store) {
+	t.Run("Save shared channel attachment", func(t *testing.T) {
+		attachment := &model.SharedChannelAttachment{
+			FileId:          model.NewId(),
+			RemoteClusterId: model.NewId(),
+		}
+
+		saved, err := ss.SharedChannel().SaveAttachment(attachment)
+		require.Nil(t, err, "couldn't save shared channel attachment", err)
+
+		require.Equal(t, attachment.FileId, saved.FileId)
+		require.Equal(t, attachment.RemoteClusterId, saved.RemoteClusterId)
+	})
+
+	t.Run("Save invalid shared channel attachment", func(t *testing.T) {
+		attachment := &model.SharedChannelAttachment{
+			FileId:          "",
+			RemoteClusterId: model.NewId(),
+		}
+
+		_, err := ss.SharedChannel().SaveAttachment(attachment)
+		require.NotNil(t, err, "should error saving invalid attachment", err)
+	})
+
+	t.Run("Save shared channel attachment with invalid remote id", func(t *testing.T) {
+		attachment := &model.SharedChannelAttachment{
+			FileId:          model.NewId(),
+			RemoteClusterId: "bogus",
+		}
+
+		_, err := ss.SharedChannel().SaveAttachment(attachment)
+		require.Error(t, err, "expected error for invalid remote id")
+	})
+}
+
+func testUpsertSharedChannelAttachment(t *testing.T, ss store.Store) {
+	t.Run("Upsert new shared channel attachment", func(t *testing.T) {
+		attachment := &model.SharedChannelAttachment{
+			FileId:          model.NewId(),
+			RemoteClusterId: model.NewId(),
+		}
+
+		_, err := ss.SharedChannel().UpsertAttachment(attachment)
+		require.NoError(t, err, "couldn't upsert shared channel attachment", err)
+
+		saved, err := ss.SharedChannel().GetAttachment(attachment.FileId, attachment.RemoteClusterId)
+		require.NoError(t, err, "couldn't get shared channel attachment", err)
+
+		require.NotZero(t, saved.CreateAt)
+		require.Equal(t, saved.CreateAt, saved.LastSyncAt)
+	})
+
+	t.Run("Upsert existing shared channel attachment", func(t *testing.T) {
+		attachment := &model.SharedChannelAttachment{
+			FileId:          model.NewId(),
+			RemoteClusterId: model.NewId(),
+		}
+
+		saved, err := ss.SharedChannel().SaveAttachment(attachment)
+		require.Nil(t, err, "couldn't save shared channel attachment", err)
+
+		// make sure enough time passed that GetMillis returns a different value
+		time.Sleep(50 * time.Millisecond)
```

Comment:
1 millisecond should be sufficient.

Reply:
Fixed via https://github.com/mattermost/mattermost-server/pull/16880/commits/0314c8a0ac14f2f34e9778c167978e56f4ec4068